### PR TITLE
feat(application-payment): Splitting the payment callback into its own controller

### PR DESCRIPTION
**Pull request by arnorhs**
This route should only really be exposed through x-road, so excluding it from the openapi, removing the user guard, and making the controller action more friendly to receiving a trusted call through an internal ingres

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/application-system/api/src/app/modules/payment/payment-callback.controller.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import type { Callback } from '@island.is/api/domains/payment'
+
+@Controller()
+export class PaymentCallbackController {
+  constructor(
+    @InjectModel(Payment)
+    private paymentModel: typeof Payment,
+  ) {}
+
+  @Post('application-payment/:application_id/:id')
```
Comment (jeremybarbet) [inline apps/application-system/api/src/app/modules/payment/payment-callback.controller.ts:13]: You should be using camel case instead of snake case

Comment (arnorhs) [reply apps/application-system/api/src/app/modules/payment/payment-callback.controller.ts:13]: excellent point, will fix

**Review by jGeir — APPROVED** [review]

File: apps/application-system/api/src/app/modules/payment/payment-callback.controller.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import type { Callback } from '@island.is/api/domains/payment'
+
+@Controller()
+export class PaymentCallbackController {
+  constructor(
+    @InjectModel(Payment)
+    private paymentModel: typeof Payment,
+  ) {}
+
+  @Post('applicationPayment/:application_id/:id')
```
[the remark this answers is not in the document]
Comment (arnorhs) [reply apps/application-system/api/src/app/modules/payment/payment-callback.controller.ts:13]: 🤦 ok, makes more sense

File: apps/application-system/api/src/app/modules/payment/payment.controller.ts [hunk 3]
```diff
@@ -91,48 +88,6 @@ export class PaymentController {
...
-        where: {
-          id,
-          application_id: applicationId,
-        },
-      },
-    )
-  }
-
   @Scopes(ApplicationScope.read)
   @Get('applications/:application_id/payment-status')
```
[the remark this answers is not in the document]
Comment (arnorhs) [reply apps/application-system/api/src/app/modules/payment/payment.controller.ts:92]: excellent, sorry about that

**Review by jeremybarbet — APPROVED** [review]
LGTM

**Pull request merged**